### PR TITLE
Remove extra thread and allow deinit of BLE

### PR DESCRIPTION
**Pull request by kasperl**
All the work done by @mikkeldamsgaard in #749.

### CODE REVIEW

File: src/event_sources/ble_esp32.cc [hunk 1]
```diff
@@ -89,7 +87,6 @@ void BLEEventSource::entry() {
         nimble_port_run();
       }
 
-      nimble_port_freertos_deinit();
```
Comment (floitsch) [inline src/event_sources/ble_esp32.cc:92]: Is this not needed anymore?

File: src/resources/ble_esp32.cc [hunk 2]
```diff
@@ -559,6 +546,9 @@ PRIMITIVE(init) {
     }
   }
 
+  group->register_resource(gap);
```
Comment (floitsch) [inline src/resources/ble_esp32.cc:549]: If the `gap` is only registered here, do we then need to deallocate it explicitly when the init-server fails (line 544/545)?

**Review by floitsch — APPROVED** [review]
LGTM.

File: src/resources/ble_esp32.cc (hunk 2 above)
Comment (kasperl) [reply src/resources/ble_esp32.cc:549]: Yeah. I believe we do need to `delete gap` before returning in line 545. Thoughts, @mikkeldamsgaard?

File: src/event_sources/ble_esp32.cc (hunk 1 above)
Comment (kasperl) [reply src/event_sources/ble_esp32.cc:92]: Yeah, it turns out that this isn't needed.

**Pull request merged**